Src/Support/Google.Apis.Auth.Tests/OAuth2/ExternalAccountCredentialTestsBase.cs

```diff
@@ -0,0 +1,174 @@
+        protected static readonly string SubjectTokenJson = $@"{{""{SubjectTokenJsonField}"": ""{SubjectTokenText}""}}";
+
+        protected const string TokenUrl = "https://dummy.token.url/";
+        protected const string GrantTypeClaim = "grant_type=urn:ietf:params:oauth:grant-type:token-exchange";
+        protected const string RequestedTokenTypeClaim = "requested_token_type=urn:ietf:params:oauth:token-type:access_token";
+        protected const string Audience = "dummy_audience";
+        protected const string SubjectTokenType = "dummy_token_type";
+        protected const string Scope = "dummy_scope";
+        protected const string ImpersonationScope = "https://www.googleapis.com/auth/iam";
+        protected const string ClientId = "dummy_client_ID";
+        protected const string ClientSecret = "dummy_client_secret";
+        protected const string WorkforcePoolUserProject = "dummy_workforce_project";
+        protected const string ImpersonationUrl = "https://dummy.impersonation.url/";
+
+        protected const string AccessToken = "dummy_access_token";
+        protected const string RefreshedAccessToken = "dummy_refreshed_access_token";
+        protected const string ImpersonatedAccessToken = "dummy_impersonated_access_token";
+        protected const string QuotaProject = "dummy_project_id";
+        protected const string QuotaProjectHeaderName = "x-goog-user-project";
+
+        protected static async Task<HttpResponseMessage> ValidateAccessTokenRequest(HttpRequestMessage accessTokenRequest, string scope, bool isWorkforce = false)
+        {
+            Assert.Equal(TokenUrl, accessTokenRequest.RequestUri.ToString());
+            Assert.Equal(HttpMethod.Post, accessTokenRequest.Method);
+
+            string contentText = WebUtility.UrlDecode(await accessTokenRequest.Content.ReadAsStringAsync());
+
+            if (isWorkforce)
+            {
+                Assert.Null(accessTokenRequest.Headers.Authorization);
+                Assert.Contains($"options={{\"userProject\":\"{WorkforcePoolUserProject}\"}}", contentText);
+            }
+            else
+            {
+                Assert.Equal("Basic", accessTokenRequest.Headers.Authorization.Scheme);
+                Assert.Equal(Convert.ToBase64String(Encoding.UTF8.GetBytes($"{ClientId}:{ClientSecret}")), accessTokenRequest.Headers.Authorization.Parameter);
+                Assert.DoesNotContain("options=", contentText);
+            }
+
+            Assert.Contains(GrantTypeClaim, contentText);
+            Assert.Contains(RequestedTokenTypeClaim, contentText);
+            Assert.Contains($"audience={Audience}", contentText);
+            Assert.Contains($"subject_token_type={SubjectTokenType}", contentText);
+            Assert.Contains($"subject_token={SubjectTokenText}", contentText);
+            Assert.Contains($"scope={scope}", contentText);
+
+            return new HttpResponseMessage
```

Comment:
I'd probably do the serialization in a separate statement above, then just `return new HttpResponseMessage { Content = new StringContent(content, Encoding.UTF8) };` - aside from anything else, that's simpler when debugging.

Reply:
Given all the code below, it might also be worth extracting `new HttpResponseMessage { Content = new StringContent(content, Encoding.UTF8) }` into a helper method.

Reply:
In fact, we might want a helper method to do everything - just given the access token and expiry, as that's all that really seems to vary.

Reply:
Done.